app/controllers/api/v1/offers_controller.rb

```diff
@@ -70,9 +70,11 @@ def index
 
       api :GET, '/v1/offers/search?searchText=xyz', "Search for offers"
       def search
-        @offers = @offers.search(search_text: params['searchText'])
-        @offers = @offers.order('created_at desc').limit(25)
-        render json: @offers.with_summary_eager_load.to_a, each_serializer: summary_serializer, summarize: true
+        records = @offers.search({ search_text: params['searchText'], states: array_param(:state) })
```

Comment:
@abulsayyad123 , there's no need of setting @offers to records.
you can directly do
@offers = @offers.search ....
@offers = @offers.order ....
.
.
so on..

Reply:
although that's true, I do believe avoiding data mutations is a good practice
`records = doSomeFilteringOf(@offers)`

I'm fine either way, given that Controllers are very standardised, so risks are limited.
But generally I'd strongly argue against it (think functional programming 😄)